### PR TITLE
Should be validating the global network policy name

### PR DESCRIPTION
**Pull request by robbrockbank**
## Description

GNP name also needs to be locked down to not include `.`

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by neiljerram** [comment]
I've assigned myself as a reviewer here, but .... let me see how clearly I can say this .... we should *ABSOLUTELY NOT* merge this before we have released the beta, so I am not even going to review it right now.

**Comment by robbrockbank** [comment]
@neiljerram :  You assigned yourself this - are you planning to review it now that the beta freeze is over?  It absolutely should go into GA.

File: lib/validator/v3/validator.go [hunk 1]
```diff
@@ -55,6 +55,9 @@ var (
 	// must be the standard name format (nameRegex) prefixed with "knp.default".
 	networkPolicyNameRegex = regexp.MustCompile("^((" + nameLabelFmt + ")|(knp\\.default\\.(" + nameSubdomainFmt + ")))$")
 
+	// GlobalNetworkPolicy names must either be a simple DNS1123 label format (nameLabelFmt).
```
Comment (neiljerram) [inline lib/validator/v3/validator.go]: Delete "either" (or add the "or ...", if there should be one).

File: lib/validator/v3/validator.go [hunk 2]
```diff
@@ -813,6 +777,70 @@ func validateNetworkPolicy(v *validator.Validate, structLevel *validator.StructL
...
+	validateObjectMetaLabels(v, structLevel, gnp.Labels)
+
+	if spec.DoNotTrack && spec.PreDNAT {
+		structLevel.ReportError(reflect.ValueOf(spec.PreDNAT),
+			"PolicySpec.PreDNAT", "", reason("PreDNAT and DoNotTrack cannot both be true, for a given PolicySpec"))
+	}
+
+	if spec.PreDNAT && len(spec.Egress) > 0 {
+		structLevel.ReportError(reflect.ValueOf(spec.Egress),
+			"PolicySpec.Egress", "", reason("PreDNAT PolicySpec cannot have any Egress"))
```
Comment (neiljerram) [inline lib/validator/v3/validator.go]: I appreciate you didn't change this, but I'd add "rules" at the end here.  (i.e. "... any Egress rules")

**Comment by robbrockbank** [comment]
@neiljerram :  Hi Neil, ta for reviewing.  I've made the two markups, squashed and pushed.  Let me know if this is good to go.

**Comment by robbrockbank** [comment]
@neiljerram :  given the two comments are minor nits I'll get Brendan to review so that I can get this merged today - hope that's ok.

**Pull request merged**